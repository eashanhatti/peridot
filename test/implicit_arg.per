define id
  : Function(implicit A : Type, A) -> A
  = function(x) => x;

define x : Bool = id(true);
