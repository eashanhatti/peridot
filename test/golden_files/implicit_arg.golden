((fromList
    [(UVGlobal 1000, Nothing),
     (UVGlobal 1001, Nothing),
     (UVGlobal 1002, Nothing),
     (UVGlobal 1003, Nothing),
     (UVGlobal 1004, Nothing),
     (UVGlobal 1005, Nothing),
     (UVGlobal 1006, Nothing),
     (UVGlobal 1007, Nothing),
     (UVGlobal 1008, Nothing),
     (UVGlobal 1009, Nothing),
     (UVGlobal 1010, Nothing),
     (UVGlobal 1011, Nothing),
     (UVGlobal 1012, Just (Rigid TwoType))],
  fromList [],
  fromList [],
  fromList []),
 Let
   (fromList
      [ObjTerm
         (Id {unId = 0})
         (ObjFunType
            Unification
            (TypeType Obj)
            (ObjFunType
               Explicit
               (LocalVar (Index {unIndex = 0}))
               (LocalVar (Index {unIndex = 1}))))
         (ObjFunIntro
            (ObjFunIntro (LocalVar (Index {unIndex = 0})))),
       ObjTerm
         (Id {unId = 1})
         (Rigid TwoType)
         (ObjFunElim
            (ObjFunElim
               (GlobalVar (Id {unId = 0})) (UniVar (UVGlobal 1012)))
            (Rigid TwoIntro0))])
   (TypeType Obj))
